lib/chef/knife/bmcs_server_list.rb

```diff
@@ -25,10 +26,39 @@ def run
         options = {}
         options[:limit] = config[:limit] if config[:limit]
 
+        servers = {}
+
         response = compute_client.list_instances(compartment_id, options)
 
-        display_list(response, ['Display Name', 'ID', 'State']) do |item|
-          [item.display_name, item.id, item.lifecycle_state]
+        if config[:format] == 'summary' && response.data
+          # build a hash of hashs, keyed off of server_id
+          vnics = compute_client.list_vnic_attachments(compartment_id, {})
```

Comment:
I re-read this comment a bunch of times, and I finally think that I understand what you're asking for...  Instead of grabbing the list of servers (in one big chunk) and then grabbing the list of vnic_attachements for the entire compartment (in one big chunk) and matching them up, you're suggesting that I should iterate over the list of servers, and call list_vnic_attachements for each server, to make sure that the returned data contains all the responses.